pkg/capacity/capacity.go

```diff
@@ -642,8 +642,10 @@ func (c *Controller) syncCapacity(ctx context.Context, item workItem) error {
 		// scenario that we end up creating two objects for the same work item, the second
 		// one will be recognized as duplicate and get deleted again once we receive it.
 	} else if capacity.Capacity.Value() == quantity.Value() &&
+		capacity.MaximumVolumeSize == maximumVolumeSize &&
+		(maximumVolumeSize == nil || capacity.MaximumVolumeSize.Value() == maximumVolumeSize.Value()) &&
```

Comment:
This gets hard to review for correctness.

Let's move this into a helper function:
```
func sizesAreEqual(a, b *resource.Quantity) bool {
   if a == b {
     // Both nil or the same value.
     return true
  }
  if a == nil {
     // b must be non-nil.
     return false
  }
  // Both not nil, compare values.
  return a.Value() == b.Value()
}
```